### ISSUE TITLE
:keys can't be used in  :ret position

### ISSUE THREAD

**Issue by ikitommi** [clj-kondo/clj-kondo#783]
**version**

clj-kondo v2020.02.15

**platform**

native

**editor**

command line

**problem**

the `:keys` works in function `:args`, but not in `:ret`:

**repro**

```bash
echo '(do (defn fun2 [m] (str (:a m))) (+ 1 (:a (fun2 {:a "1"}))))' | clj-kondo --lint - --config '{:linters {:type-mismatch {:namespaces {user {fun2 {:arities {1 {:args [{:op :keys, :req {:a :int}}], :ret {:op :keys, :req {:a :string}}}}}}}}}}'
<stdin>:1:53: error: Expected: integer, received: string.
linting took 14ms, errors: 1, warnings: 0
```

**expected behavior**

... the `:ret` should be identified as map with :a key of type `:string` and the `:a` lookup on it should return a `:string`.

**Comment by borkdude** [comment]
To be clear, you expect this?

```
(+ 1 (:a (fun2 {:a "1"})))
       ^ ;; Expected: number, received: string
```

**Comment by ikitommi** [comment]
Yes, that.

**Comment by borkdude** [comment]
Thanks. I added the clj-together label. If this project gets funded by Clojurists Together I can do it during the funding period.
